Add create and patch methods

lucassouto opened the pull request.

Commit:
feat(secrets): Add create and patch methods

Commit:
docs(provider): Add new methods docs

Commit:
chore(nameko-vault): Add pyproject.toml

Commit:
chore(nameko-vault): Remove setup.py

Commit:
chore(CI): Add github actions

Commit:
style(black): Apply black format

Commit:
chore(CI): Publish package in CI

Commit:
tests(provider): Add tests create and patch method

Commit:
chore(CI): Add test step

oesgalha commented on README.md:
```diff
@@ -69,3 +69,55 @@ vault.get_kv_secret(mount_point="example", path="path/test")
+   'lease_id': '',
+   'renewable': False,
+   'lease_duration': 0,
+   'data': {
```
Is this example correct? Looking to the other example in the README this one looks like the "metadata", not the "data".

lucassouto replied:
That's correct, the `create_or_update` and` patch` methods doesn't return a "metadata" key

oesgalha replied:
Ok, thanks for confirming

oesgalha commented on README.md:
```diff
@@ -69,3 +69,55 @@ vault.get_kv_secret(mount_point="example", path="path/test")
+   'lease_id': '',
+   'renewable': False,
+   'lease_duration': 0,
+   'data': {
```
Ditto.
If this returns the data (plus the metadata) it would be nice to show how the patch method behaves: changing only the "example" key in this demonstration.

lucassouto replied:
It does not return data, I believe that the ideal was for this key to be "metadata", but the SDK uses "data". 

This example is just changing the secret "example" key:
`{"example": "Test", "number": 42}` -> `{"example": "New Test", "number": 42}`

oesgalha reviewed (changes requested):
Added some questions regarding the docs

Commit:
chore(package): Remove lockfile

- From Poetry docs:
`For libraries it is not necessary to commit the lock file.`

https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control

Here is another source about when you should keep lockfiles in (or out)
of version control:
https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

Commit:
chore(package): Update package version

- Change package version

- Change author email

Commit:
chore(CI): Change publishing method to use token

oesgalha committed:
Merge pull request #4 from instruct-br/chore/publish-package

Creates CI/CD pipeline and project configuration with Poetry

oesgalha reviewed (approved):
:shipit:
